src/services/upload/exifService.ts

```diff
@@ -34,17 +35,31 @@ export async function getExifData(
 }
 
 function getUNIXTime(exifData: any) {
-    const dateString: string = exifData.DateTimeOriginal || exifData.DateTime;
-    if (!dateString || dateString === '0000:00:00 00:00:00') {
+    try {
+        if (!exifData.DateTimeOriginal) {
+            throw Error(CustomError.MISSING_EXIF_TAG);
+        }
+        if (exifData.DateTimeOriginal instanceof Date) {
+            return exifData.DateTimeOriginal.getTime() * 1000;
+        } else if (typeof exifData.DateTimeOriginal === 'string') {
+            const dateString = exifData.DateTimeOriginal;
+            if (dateString === '0000:00:00 00:00:00') {
+                throw Error(CustomError.MISSING_EXIF_TAG);
+            }
+            const parts = dateString.split(' ')[0].split(':');
+            const date = new Date(
+                Number(parts[0]),
+                Number(parts[1]) - 1,
+                Number(parts[2])
+            );
+            return date.getTime() * 1000;
+        } else {
+            throw Error(CustomError.MISSING_EXIF_TAG);
+        }
+    } catch (e) {
+        logError(e, 'getUnixTime failed', { exifData });
```

Comment:
Shall we avoid explicitly logging `exifData` data as it would contain some private information about the photo?  If we feel the stack trace or exception isn't enough for debugging, we can plan to add it later.